pkg/image/controller/trigger/annotations.go

```diff
@@ -0,0 +1,171 @@
+package trigger
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	ometa "github.com/openshift/origin/pkg/api/meta"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/api/v1/trigger"
+)
+
+const triggerAnnotationKey = "image.openshift.io/triggers"
```

Comment:
We should be able to support different kinds of triggers with this controller, no?